### ISSUE TITLE
Not working on latest PhpStorm version (2023.3.6)

### ISSUE THREAD

**Issue by Misairuzame** [semgrep/semgrep-intellij#52]
Hi,
We are testing Semgrep and its extension for the latest version of PhpStorm (2023.3.6) but, as stated in previous issues (https://github.com/semgrep/semgrep-intellij/issues/24 and https://github.com/semgrep/semgrep-intellij/issues/31), the extension does nothing when clicking on any of the options of its dropdown menu. Semgrep on the command line, however, works perfectly.

Is this extension still maintained, as the problem seems to be approx. 6 months old? Should we just rely on Semgrep's command line?
Thank you for your time.

**Comment by ajbt200128** [comment]
Hello! This extension is still being maintained. What version of semgrep are you using?

**Comment by Misairuzame** [comment]
Hi,
Thank you for your quick reply.
I'm using semgrep 1.67.0 installed via pip.

I just tried using the extension again, and I can see these kind of errors in PhpStorm's logs, maybe they can be useful:
```2024-04-04 17:28:40,020 [  43234]   WARN - #c.i.o.a.e.ActionUtil - Action is not performed because target component is not showing: action=com.semgrep.idea.actions.ScanWorkspaceFullAction, component=com.intellij.openapi.editor.impl.EditorComponentImpl
2024-04-04 17:29:43,020 [ 106234]   WARN - #c.i.o.a.e.ActionUtil - Action is not performed because target component is not showing: action=com.semgrep.idea.actions.ScanWorkspaceAction, component=com.intellij.openapi.editor.impl.EditorComponentImpl
2024-04-04 17:29:46,270 [ 109484]   WARN - #c.i.o.a.e.ActionUtil - Action is not performed because target component is not showing: action=com.semgrep.idea.actions.RefreshRulesAction, component=com.intellij.openapi.editor.impl.EditorComponentImpl
2024-04-04 17:29:49,460 [ 112674]   WARN - #c.i.o.a.e.ActionUtil - Action is not performed because target component is not showing: action=com.semgrep.idea.actions.LoginAction, component=com.intellij.openapi.editor.impl.EditorComponentImpl
2024-04-04 17:29:53,189 [ 116403]   WARN - #c.i.o.a.e.ActionUtil - Action is not performed because target component is not showing: action=com.semgrep.idea.actions.LogoutAction, component=com.intellij.openapi.editor.impl.EditorComponentImpl```

**Comment by ajbt200128** [comment]
can you follow this and see if there's any logs from lsp https://plugins.jetbrains.com/docs/intellij/language-server-protocol.html#troubleshooting (going to update the bug report template soon to ask users to do this, sorry about that)

**Comment by ajbt200128** [comment]
Can't seem to reproduce this issue. Is this a large repository (i.e. few thousand files)? If so the scan workspace command may timeout, and you can try bumping the # of jobs in the semgrep plugin settings. If you paste example code from a rule you're expecting to be found into an existing file, does any error show up?

**Comment by Misairuzame** [comment]
Hi, sorry for the delayed response. The repository is definitely not that large; it's just a clone of https://github.com/digininja/DVWA. Once again none of the actions I try to perform produce any kind of output - for example, I can't even sign in. Looking at a piece of code with a SQL Injection vulerability that I know for a fact is found by a terminal semgrep scan, I can't see any error in PhpStorm.
Just FYI, I was able to run the Semgrep extension successfully maybe once, about a month ago. After that, when I reopened the IDE, I was never able to use it again.

**Comment by ajbt200128** [comment]
Can you try going into the semgrep settings and unchecking "Only Git Dirty", restarting and seeing if that works? It may take a few extra seconds to start up. The extensions only scan files that have changed since the last commit, so if you clone a repository, and then open a file, there will be no findings since nothing has changed since the last commit. This functionality is very similar to pre-commit

**Comment by Misairuzame** [comment]
Hello, I admit I'm not sure where to find that setting. I tried updating Semgrep itself via pip (to 1.70.0), PhpStorm (to 2024.1.1) and the Semgrep plugin (to 0.2.4). I was messing around with the extension and it was behaving as usual on my machine, e.g., not responding to any action. After some time I tried again the option "Sign in with Semgrep" and the extension - for the first time - showed a popup saying I was already logged in. I then tried to do a full scan (including unmodified files) and magically, the scan worked and I can see issues being highlighted directly in the code.
Not sure what fixed it, but it seems to be at least somewhat working now.
Thank you for your continued help!
